lib/modules.nix

```diff
@@ -232,19 +235,19 @@ let
 
         config = {
           _module.args = {
-            inherit extendModules class;
+            inherit extendModules;
             moduleType = type;
           };
-          _module.specialArgs = specialArgs;
+          _module.specialArgs = extendedSpecialArgs;
         };
       };
 
       merged =
         let collected = collectModules
           class
-          (specialArgs.modulesPath or "")
+          (extendedSpecialArgs.modulesPath or "")
           (regularModules ++ [ internalModule ])
-          ({ inherit lib options config specialArgs; } // specialArgs);
+          ({ inherit lib options config; specialArgs = extendedSpecialArgs; } // extendedSpecialArgs );
```

Comment:
This is a very literal interpretation of _add to specialArgs_. That's kinda ok, but not required.
What you can see (mostly) here is that _module arguments_ = _built-in arguments_ + _special args_ +  _\_module.args_.

So instead of making `_class` a special arg, we could make it a built-in arg, saving us the `extendedSpecialArgs` binding, a `//`, and most of all, it will be a bit simpler and IMO more consistent.

Reply:
Is that what you meant?